docs/code/MlNetCookBook.md

```diff
@@ -45,25 +48,19 @@ This is because we rely on extension methods a lot, and they only become availab
 We are still re-organizing the namespaces, and trying to improve the story. In the meantime, the following namespaces prove useful for extension methods:
 ```csharp
 using Microsoft.ML.Data;
-using Microsoft.ML.StaticPipe;
 using Microsoft.ML.Trainers;
 using Microsoft.ML.Transforms;
 ```
 
-- *Why is there two ways of doing things? Which one is better, static or dynamic?*
-We don't know yet, that's why we have two ways! 
+- *What is the [MLContext](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.mlcontext?view=ml-dotnet)?*
+The MLContext is a starting point for all ML.NET operations. 
+It is instantiated by user, and provides mechanisms for logging, exception tracking and logging, setting the source of randomness.
+It is the entry points for training, prediction, model operations and also serves as a catalog of available operations.
+You will need one MlContext object for your pipelines or inference code. 
 
-If you are loading an existing model from a stream, there's no need to use static types (and it's also pretty hard to do). Also, if the data view's schema is only known at runtime, there is no way to use static types. In other scenarios we tend to prefer the static types, since this way gives you compiler support: it's more likely that, if your code compiles, it will also work as intended.
-
-- *Why do we call `reader.MakeNewEstimator` to create a pipeline?*
-In the static pipeline, we need to know the two 'schema' types: the input and the output to the pipeline.
-One of them is already known: typically, the output schema of `reader` (which is the same as the schema of `reader.Read()`) is also the input schema of the learning pipeline. 
-
-The call to `x.MakeNewEstimator` is only using the `x`'s *schema* to create an empty pipeline, it doesn't use anything else from `x`. So, the following three lines would create the exactly same (empty) pipeline:
 ```csharp
-var p1 = reader.MakeNewEstimator();
-var p2 = reader.Read(dataLocation).MakeNewEstimator();
-var p3 = p1.MakeNewEstimator();
+// as a catalog of available operations and as the source of randomness.
+var mlContext = new MLContext();
```

Comment:
Was this line supposed to be kept?

Reply:
yes, since we are talking about the MLContext. 

---
In reply to: [251650454](https://github.com/dotnet/machinelearning/pull/2284#discussion_r251650454) [](ancestors = 251650454)

Reply:
It seems this was the second part of a two line comment, and it feels like it's missing something. Maybe I would add the first part of the sentence.

Could you capitalize otherwise?

---
In reply to: [251944224](https://github.com/dotnet/machinelearning/pull/2284#discussion_r251944224) [](ancestors = 251944224,251650454)